### PR TITLE
Added simpliciality measures

### PR DESCRIPTION
**Pull request by nwlandry**
This PR adds all the functionality from this [repo](https://github.com/nwlandry/the-simpliciality-of-higher-order-networks).

### CODE REVIEW

File: xgi/algorithms/simpliciality.py [hunk 1]
```diff
@@ -0,0 +1,434 @@
...
+    sub_edges = list(_powerset(face, min_size=min_size, max_size=len(face) - 1))
+    count = 0
+    for e in sub_edges:
+        if t.search(e):
+            count += 1
+
+    return count
+
+
+def _count_subfaces(t, face, min_size=1):
```
Comment (tlarock) [inline xgi/algorithms/simpliciality.py:346]: Is this a repeat of the above function with the same signature?

File: xgi/algorithms/simpliciality.py [hunk 2]
```diff
@@ -0,0 +1,434 @@
...
+                    # faces so no inclusions.
+                    if not t.search(c):
+                        redundant_missing_faces.add(frozenset(c))
+
+        nm = _max_number_of_subfaces(min_size, len(e))
+        nf = _count_subfaces(t, e, min_size)
+        rmf = len(redundant_missing_faces)
+        ms += nm - nf - rmf
+
+    try:
```
Comment (tlarock) [inline xgi/algorithms/simpliciality.py:79]: Do we have a style preference for a try/catch block here rather than checking for 0 denominator in a normal conditional? Since `s` and `ms` should both be integers, could we just check `if (ms+s) > 0` rather than relying on exception handling?

Comment (tlarock) [reply xgi/algorithms/simpliciality.py:79]: I think this comment applies to the returns of all of these functions. Practically the outcome should be the same using either approach, so in my view it is a style/readability question. My personal preference is for a conditional, but open to leaving it as is if it is consistent with style elsewhere or desireable for another reason.

**Review by tlarock — CHANGES_REQUESTED** [review]
Looks good overall and tests appear to pass. Worth considering how we structure the return of the functions, since using exception handling rather than conditionals is a design choice. One function appears to be duplicated. Otherwise good to go!

**Comment by nwlandry** [comment]
Thanks for the review, @tlarock! Someone pointed out a minor error in my simpliciality paper, so I'm determining what to do with that first before making your suggested changes.

File: xgi/algorithms/simpliciality.py (hunk 2 above)
Comment (nwlandry) [reply xgi/algorithms/simpliciality.py:79]: Great point. I fixed this according to your suggestion.

File: xgi/algorithms/simpliciality.py (hunk 1 above)
Comment (nwlandry) [reply xgi/algorithms/simpliciality.py:346]: Oops --- great catch. Fixed now!

**Comment by nwlandry** [comment]
Hi @tlarock --- I believe I addresed all of your comments. The URL checker fails, but this was addressed in another PR so it will go away once we merge into the main branch.

File: xgi/algorithms/simpliciality.py [hunk 3]
```diff
@@ -0,0 +1,369 @@
...
+    Parameters
+    ----------
+    H : xgi.Hypergraph
+        The hypergraph of interest
+    min_size: int, default: 2
+        The minimum hyperedge size to include when
+        calculating whether a hyperedge is a simplex
+        by counting subfaces.
+    exclude_min_size : bool, optional
+        Whether to include minimal simplices when counting simplices, by default True
```
Comment (tlarock) [inline xgi/algorithms/simpliciality.py:70]: The name is `exclude_min_size`, but the documentation reads "Whether to _include_...default `True`", which makes my head spin a bit! Maybe reword for clarity?

File: xgi/algorithms/simpliciality.py [hunk 4]
```diff
@@ -0,0 +1,369 @@
...
+    "The simpliciality of higher-order order networks"
+    by Nicholas Landry, Jean-Gabriel Young, and Nicole Eikmeier,
+    *EPJ Data Science* **13**, 17 (2024).
+    """
+    return 1 - simplicial_edit_distance(
+        H, min_size=min_size, exclude_min_size=exclude_min_size
+    )
+
+
+def simplicial_edit_distance(H, min_size=2, exclude_min_size=True, normalize=True):
```
Comment (tlarock) [inline xgi/algorithms/simpliciality.py:55]: I don't think I could fully understand from the documentation alone what exactly the parameters `min_size` and `exclude_min_size` control. I had to go read the code, especially for the latter parameter. I'm not sure if there is a straightforward solution without either (a) having longer more expressive parameter names, or (b) a more thorough description in the main docstring that explains explicitly how they change the calculation, but maybe something worth thinking about a bit since I am somewhat familiar with the measures and couldn't quite understand.

**Review by tlarock — COMMENTED** [review]
Very close! Sorry to be a pain about the documentation, but hopefully easier to clairfy now than later :) Of course feel free to take or leave

File: xgi/algorithms/simpliciality.py (hunk 4 above)
Comment (nwlandry) [reply xgi/algorithms/simpliciality.py:55]: On it! Thanks for bringing this up!

File: xgi/algorithms/simpliciality.py (hunk 3 above)
Comment (nwlandry) [reply xgi/algorithms/simpliciality.py:70]: Good call. I will correct everywhere.

**Review by tlarock — APPROVED** [review]
